Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.Core.UnitTests/API.Tests/TeamsControllerTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Microsoft.Teams.App.KronosWfc.Common;
+using Microsoft.Teams.Shifts.Integration.API.Common;
+using Microsoft.Teams.Shifts.Integration.API.Controllers;
+using Microsoft.Teams.Shifts.Integration.BusinessLogic.Models;
+using Microsoft.Teams.Shifts.Integration.BusinessLogic.Providers;
+using Microsoft.Teams.Shifts.Integration.BusinessLogic.ResponseModels;
+using Microsoft.Teams.Shifts.Integration.Core.UnitTests.Common;
+using NSubstitute;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Teams.Shifts.Integration.Core.UnitTests.API.Tests
+{
+    public class TeamsControllerTests
+    {
+        /// <summary>
+        /// This unit test mimics the behaviour of cancelling an open shift request. The ProcessOpenShiftRequest method is private
+        /// meaning we had to by hand map out the logic of this condition which is not ideal. However, one level higher in the stack 
+        /// was also not ideal as that is the public sync method and would involve more mocking than would be worth for this test case.
+        /// </summary>
+        /// <param name="id">ID of entity to retract</param>
+        /// <param name="allOpenShiftRequestMappingEntity">The open shift request.</param>
+        /// <param name="openShiftRequestMappingEntityProvider">The provider with logic for open shift requests.</param>
+        /// <param name="openShiftController">Open shift controller business logic.</param>
+        [Theory]
+        [InlineAutoNSubstituteData()]
```

Comment:
This is a bit nitpicky, but because you aren't providing any inline data to pass into the test, just [AutoNSubstituteData] would be the correct attribute to use here.

Reply:
Fixed!